cmake/build_common.cmake

```diff
@@ -0,0 +1,235 @@
+if(NOT DEFINED ROOT OR NOT DEFINED ARCH)
+    message(
+	FATAL_ERROR 
+	    "Assert: ROOT = ${ROOT}; ARCH = ${ARCH}"
+    )
+endif()
+
+include("${CMAKE_CURRENT_LIST_DIR}/Version.cmake")
+
+set(
+    BUILD
+	0
+)
+
+if(DEFINED ENV{BUILD_NUMBER})
+    set(
+	BUILD
+	    $ENV{BUILD_NUMBER}
+    )
+endif()
+
+set(
+    TAG
+	""
+)
+
+if(DEFINED ENV{TAG})
+    set(
+	TAG
+	    "$ENV{TAG}"
+    )
+endif()
+
+set(
+    BUILD_PATH
+	"${CMAKE_CURRENT_LIST_DIR}/../build"
+)
+
+set(
+    PACKAGE_NAME
+	"cryptopp-${VERSION}-${ARCH}-${BUILD}${TAG}"
+)
+
+set(
+    CMAKE_INSTALL_PREFIX
+	${ROOT}/${PACKAGE_NAME}
+)
+
+set(
+    GNU_RELEASE_LIBS_NAME
+	"libcryptopp"
+)
+
+set(
+    GNU_DEBUG_LIBS_NAME
+	${GNU_RELEASE_LIBS_NAME}d
+)
+
+set(
+    MSVS_RELEASE_LIBS_NAME
+	"cryptlib"
+)
+
+set(
+    MSVS_DEBUG_LIBS_NAME
+	${MSVS_RELEASE_LIBS_NAME}d
+)
+
+if(WIN32)
+    if(NOT ${ARCH} MATCHES x86)
```

Comment:
Move this variable to dedicated build scripts build_amd64.cmake and build_x86.cmake

Reply:
done